generator/integration_tests/golden/internal/golden_kitchen_sink_rest_stub.cc

```diff
@@ -0,0 +1,118 @@
+// limitations under the License.
+
+// Generated by the Codegen C++ plugin.
+// If you make any local changes, they will be lost.
+// source: generator/integration_tests/test.proto
+
+#include "generator/integration_tests/golden/internal/golden_kitchen_sink_rest_stub.h"
+#include "google/cloud/internal/absl_str_cat_quiet.h"
+#include "google/cloud/internal/rest_stub_helpers.h"
+#include "google/cloud/status_or.h"
+#include <generator/integration_tests/test.pb.h>
+#include <memory>
+
+namespace google {
+namespace cloud {
+namespace golden_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+DefaultGoldenKitchenSinkRestStub::DefaultGoldenKitchenSinkRestStub(Options options)
+    : rest_client_(rest_internal::MakePooledRestClient(
+          "https://goldenkitchensink.googleapis.com", options)),
+      options_(std::move(options)) {}
+
+DefaultGoldenKitchenSinkRestStub::DefaultGoldenKitchenSinkRestStub(
+    std::shared_ptr<rest_internal::RestClient> rest_client, Options options)
+    : rest_client_(std::move(rest_client)), options_(std::move(options)) {}
+
+StatusOr<google::test::admin::database::v1::GenerateAccessTokenResponse>
+DefaultGoldenKitchenSinkRestStub::GenerateAccessToken(
+      rest_internal::RestContext& rest_context,
+      google::test::admin::database::v1::GenerateAccessTokenRequest const& request) {
+  return rest_internal::Post<google::test::admin::database::v1::GenerateAccessTokenResponse>(
+      *rest_client_, rest_context, request,
+      absl::StrCat("/v1/",request.name(),":generateAccessToken"));
+}
+
+StatusOr<google::test::admin::database::v1::GenerateIdTokenResponse>
+DefaultGoldenKitchenSinkRestStub::GenerateIdToken(
+      rest_internal::RestContext& rest_context,
+      google::test::admin::database::v1::GenerateIdTokenRequest const& request) {
+  return rest_internal::Post<google::test::admin::database::v1::GenerateIdTokenResponse>(
+      *rest_client_, rest_context, request,
+      "/v1/token:generate");
+}
+
+StatusOr<google::test::admin::database::v1::WriteLogEntriesResponse>
+DefaultGoldenKitchenSinkRestStub::WriteLogEntries(
+      rest_internal::RestContext& rest_context,
+      google::test::admin::database::v1::WriteLogEntriesRequest const& request) {
+  return rest_internal::Post<google::test::admin::database::v1::WriteLogEntriesResponse>(
+      *rest_client_, rest_context, request,
+      "/v2/entries:write");
+}
+
+StatusOr<google::test::admin::database::v1::ListLogsResponse>
+DefaultGoldenKitchenSinkRestStub::ListLogs(
+      rest_internal::RestContext& rest_context,
+      google::test::admin::database::v1::ListLogsRequest const& request) {
+  return rest_internal::Get<google::test::admin::database::v1::ListLogsResponse>(
+      *rest_client_, rest_context, request,
+      absl::StrCat("/v2/",request.parent(),"/logs"),
+{std::make_pair("page_size", std::to_string(request.page_size())),
+std::make_pair("page_token", request.page_token())});
```

Comment:
Can we take any reasonable measures to get the initial spacing and indentation right here (`method_rest_path` and `method_http_query_parameters`)?